lib/PreviewProvider.js

```diff
@@ -342,18 +295,16 @@ PreviewProvider.prototype = {
         let responseJson = yield response.json();
         this._tabTracker.handlePerformanceEvent(event, "embedlyProxyRequestReceivedCount", responseJson.urls.length);
         this._tabTracker.handlePerformanceEvent(event, "embedlyProxyRequestSucess", 1);
-        let currentTime = Date.now();
-        newLinks.forEach(link => {
+
+        for (let link of newLinks) {
           let data = responseJson.urls[link.sanitized_url];
           if (!data) {
             return;
           }
-          ss.storage.embedlyData[link.cache_key] = Object.assign({}, ss.storage.embedlyData[link.cache_key], data);
-          if (updateAccessTime) {
-            ss.storage.embedlyData[link.cache_key].accessTime = currentTime;
-          }
-          ss.storage.embedlyData[link.cache_key].refreshTime = currentTime;
-        });
+          // Save the entry in the metadata database
+          let entry = Object.assign({}, link, {expired_at: (this.options.metadataTTL) + Date.now()});
+          this._metadataStore.asyncInsert([Object.assign({}, data, entry)]);
```

Comment:
It's preferred to do a batch insert here to improve the write performance as it has done the unique key deduping in advance. For instance,
```js
linksToInsert = newLinks.filter(link => {// filter out bad links here});
this._metadataStore.asyncInsert(linksToInsert);
```
Under the hood, the `asyncInsert()` takes an array of link objects, and inserts them into the metadata store in *one* transaction.